v2/client.go

```diff
@@ -382,10 +415,20 @@ func (c *client) Write(bp BatchPoints) error {
 	u := c.url
 	u.Path = path.Join(u.Path, "write")
 
-	req, err := http.NewRequest("POST", u.String(), &b)
+	var rd io.Reader
+	if c.encoding == GzipEncoding {
+		rd = gzipReader(&b)
```

Comment:
Instead of using a pipe, it would be better to wrap the writer itself. So above in the code where we see:

```go
var b bytes.Buffer

write := func(w io.Writer) error {
    if c.encoding == GzipEncoding {
        gzw := gzip.NewWriter(w)
        defer gzw.Close()
        w = gzipW
    }
    // normal write code
}
if err := write(&b); err != nil {
    return err
}
```

This will likely perform better than using a pipe and it also has the benefits of not copying the buffer a second time like the first implementation.